### PR TITLE
Initial Datadog dashboard charts pixlet app

### PR DESCRIPTION
**Pull request by gabe-ochoa**
# Description
A small app for showing Datadog charts! It currently only shows the first chart of a selected dashboard. I have more improvements planned but wanted to get the initial working version going.

A preview since you need API credentials configured to pull chart data:

datadog_charts

# Copilot

copilot:all

### CODE REVIEW

**Comment by gabe-ochoa** [comment]
I have read the CLA Document and I hereby sign the CLA

File: apps/datadogcharts/datadog_charts.star
Comment (matslina) [inline apps/datadogcharts/datadog_charts.star:11]:
```suggestion
load("schema.star", "schema")
load("time.star", "time")
```

File: apps/datadogcharts/datadog_charts.star
Comment (matslina) [inline apps/datadogcharts/datadog_charts.star:160]:
```suggestion
    return render.Root(child = root)

def get_schema():
    return schema.Schema(
        version = "1",
        fields = [
            schema.Text(
                id = "api_key",
                name = "API Key",
                desc = "DataDog API Key",
                icon = "key",
            ),
            schema.Text(
                id = "app_key",
                name = "App Key",
                desc = "DataDog App Key",
                icon = "key",
            ),
            schema.Text(
                id = "dashboard_id",
                name = "Dashboard ID",
                desc = "DataDog Dashboard ID",
                icon = "key",
            ),
        ],
    )

```

Comment (matslina) [reply apps/datadogcharts/datadog_charts.star:160]: This adds a schema to the app, so that users can configured it in the Tidbyt iOS/Android app. You can also test it out locally via `pixlet serve apps/datadogcharts/datadog_chart.star`.

File: apps/datadogcharts/datadog_charts.star
Comment (matslina) [inline apps/datadogcharts/datadog_charts.star:38]: Not sure if it's needed or not, but note that you can pass `ttl_seconds=4711` to cache the response. In case datadog prefers less traffic. The HTTP module should also respect cache headers by default, so that may be sufficient.

File: apps/datadogcharts/datadog_charts.star
Comment (matslina) [inline apps/datadogcharts/datadog_charts.star:33]: Not critical right now, but later on it'd be great if the app could show a default graphic with mock data in this case. That's how we generate previews for apps, letting users know a bit about what to expect before they start digging up api keys and such.

**Review by matslina — CHANGES_REQUESTED** [review]
Hey Gabe 👋!! Very, very cool! Kinda makes me want to switch to datadog. =)

File: apps/datadogcharts/datadog_charts.star
Comment (gabe-ochoa) [reply apps/datadogcharts/datadog_charts.star:33]: Happy to add it to the list for the second round of updates which I'm planning to work on next week.

File: apps/datadogcharts/datadog_charts.star
Comment (gabe-ochoa) [reply apps/datadogcharts/datadog_charts.star:39]: That's a lot easier of a caching solution that I was playing with before! I'll add it.

**Comment by matslina** [comment]
LMK when ready to merge, in case you want to get the TTL change in before.

**Comment by gabe-ochoa** [comment]
Added the TTL change and some quick fake data :) I'll merge once the build is green @matslina

**Comment by gabe-ochoa** [comment]
@matslina build is green, just need an approval.

**Review by matslina — APPROVED** [review]

**Pull request merged**
